### PR TITLE
:sparkles: Add new `custom-cache-suffix` option

### LINKED ISSUE #234: Better cache busting when using `dependency-versions: 'highest'`
## Better cache busting when using `highest` dependency versions
Using `dependency-versions: 'highest'` is the equivalent to running `composer update`, ensuring the latest versions of these dependencies are always used.

When it comes to cache keys, the key is unaware of when updates are available since the `composer.(json|lock)` files (hashed and used as part of the cache key) are not always updated to reflect these changes (the benefit of using highest is you don't have to).

## Background/problem

If a project goes a long time without updating their `composer.(json|lock)` files, it's possible that the cache restored could become really outdated This reduces the benefit of the cache since `update` will need to pull new versions every workflow run.

## Proposal/solution
One way to fix this could be to include a date in the cache key when using `highest`. In WordPress/wordpress-develop, we chose to use the [date for the previous Monday](https://github.com/WordPress/wordpress-develop/blob/fe58b3e518f729cacf7d416ea1a20eb4f00e8673/.github/workflows/phpunit-tests.yml#L120-L136) to ensure the cache was refreshed at least once a week.

### PR DESCRIPTION
**Pull request by jrfnl**
## Description

As discussed in #234.

This adds a new `custom-cache-suffix` option to the action which allows for adding a custom addition to the cache key, which can be helpful to break out of outdated caches, especially when a project does not have a committed `composer.lock` file.

Fixes #234

## Motivation and context
See #234 

## How has this been tested?

Both unit tests as well as integration tests are included with the PR.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## PR checklist
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING.md** document.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -148,6 +148,32 @@ wish to update the cache.
...
+even more specific, you can specify a suffix to be added to the cache key via the
+`custom-cache-suffix` parameter.
+
+```yaml
+# Adds a suffix to the cache key which is equivalent to the full date-time
+# of "last Monday 00:00", which means that the cache will be force refreshed
+# via the first workflow which is run every Monday.
+- uses: "ramsey/composer-install@v2"
+  with:
+    custom-cache-suffix: $(/bin/date -u --date='last Mon' "+%F")
```
Comment (ramsey) [inline README.md:172]: 🎉

**Review by ramsey — APPROVED** [review]
This is great. Thank you!

**Comment by ramsey** [comment]
Something about the [cache_key_08.exp](https://github.com/ramsey/composer-install/pull/239#diff-925f86f7996c9289363be43e9a63d34152f68370107d0018f4eddfc67f6cb089) test is causing a failure, but I can't tell what it is yet.

**Comment by jrfnl** [comment]
> Something about the [cache_key_08.exp](https://github.com/ramsey/composer-install/pull/239#diff-925f86f7996c9289363be43e9a63d34152f68370107d0018f4eddfc67f6cb089) test is causing a failure, but I can't tell what it is yet.

Windows and file permissions. Should be fixed now.

**Comment by jrfnl** [comment]
Hmm.. Composer v1 still giving trouble...

**Comment by jrfnl** [comment]
(which is weird as I had a passing build before I tidied up and rebased)

**Comment by jrfnl** [comment]
@ramsey As by now it's not the new test which is failing, could you restart the build just to check it wasn't some Composer hick-up ?

**Comment by ramsey** [comment]
Re-running

**Comment by jrfnl** [comment]
Right first round done: applied this in some 40 repos ;-)

For anyone looking for some more date-based example code:
```yml
      - name: Install Composer dependencies
        uses: ramsey/composer-install@v2
        with:
          # Bust the cache at least once every three months - output format: YYYYqQ, example: 2022q4.
          custom-cache-suffix: $(/bin/date +%Y)q$(( ($(/bin/date +%-m)-1)/3+1 ))

      - name: Install Composer dependencies
        uses: ramsey/composer-install@v2
        with:
          # Bust the cache at least once a month - output format: YYYY-MM-DD.
          custom-cache-suffix: $(date -u -d "-0 month -$(($(date +%d)-1)) days" "+%F")

      - name: Install Composer dependencies
        uses: ramsey/composer-install@v2
        with:
          # Bust the cache at least once a week - output format: YYYY-MM-DD.
          custom-cache-suffix: $(/bin/date -u --date='last Mon' "+%F")
```

**Comment by jrfnl** [comment]
Simplified version of the "once a month" code sample:

```yml
      - name: Install Composer dependencies
        uses: ramsey/composer-install@v2
        with:
          # Bust the cache at least once a month - output format: YYYY-MM.
          custom-cache-suffix: $(date -u "+%Y-%m")
```

**Pull request merged**